packages/@uppy/core/src/Uppy.ts

```diff
@@ -2280,7 +2280,7 @@ export class Uppy<
         (file) => file.progress.uploadStarted == null,
       )
 
-      if (!hasNewFiles) {
+      if (!hasNewFiles.length) {
```

Comment:
Since the name already implies it's a boolean probably better to add the `.length > 0` to the variable

```suggestion
      if (!hasNewFiles) {
```